sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -720,9 +742,12 @@ class RocksDBSuite extends AlsoTestWithChangelogCheckpointingEnabled with Shared
 
         // Another thread should not be able to load while current thread is using it
         db.load(2)
-        intercept[IllegalStateException] {
+        ex = intercept[SparkException] {
           ThreadUtils.runInNewThread("concurrent-test-thread-2") { db.load(2) }
         }
+        // Assert that the error message contains the stack trace
+        assert(ex.getMessage.contains("Thread holding the lock has trace:"))
+        assert(ex.getMessage.contains("runInNewThread"))
```

Comment:
Use `checkError`

Reply:
I'm not sure how to validate the error thrown for unreleasedThreadError given the parameters are <loggingId>, <newAcquiredThreadInfo>, <acquiredThreadInfo>, <timeWaitedMs>, and <stackTraceOutput>. Perhaps this error shouldn't be migrated to the new error framework at all?

Reply:
`checkError` suports regexp in messageParameters, BTW.

Reply:
I didn't know that thanks!